### PR TITLE
Use 'more' module in git webhook

### LINKED ISSUE #236: Some GitHub actions spam channels
Some actions on GitHub cause begiak to flood the pertinent channels with messages, like opening a review on a pull request:

```
22:24 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: Looking at this again, perhaps we can combine pairs and chains. Pairs acts normally if there's no "src" and lists the possible destination languages (in the same format) if it does. Thoughts? Just want to have as simple of an API as possible :) https://github.com/goavki/apertium-apy/pull/43#discussion_r94904572
22:24 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: Can you add a comment for each one that explains the semantics of the data structure? i.e. what are the keys and values. https://github.com/goavki/apertium-apy/pull/43#discussion_r94903868
22:25 < sushain> sigh
22:25 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: can you make it clear here whether high weight will lead to this edge being picked or low weight? (in the comment) https://github.com/goavki/apertium-apy/pull/43#discussion_r94904270
22:25 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: Clearly, we need an `in=`. https://github.com/goavki/apertium-apy/pull/43#discussion_r94904757
22:25 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: http://stackoverflow.com/questions/4613000/what-is-the-cls-variable-used-in-python-classes Nice, learned something today! https://github.com/goavki/apertium-apy/pull/43#discussion_r94903997
22:25 < begiak> apertium-apy: sushain97 * review comment created on pull request #43: Is "src" what the other endpoints use or is there a more standard term we've been using? https://github.com/goavki/apertium-apy/pull/43#discussion_r94904352
```

Or synchronizing a pull request:
```
[00:52:00] <begiak> apertium-html-tools: sushain97 * pull request #69 "Website translation" synchronize  https://github.com/goavki/apertium-html-tools/pull/69
[00:52:03] <begiak> apertium-html-tools: sushain97 * 00768c4: : assets/js/translator.js, index.html.in: Added User interface for website translation. https://github.com/goavki/apertium-html-tools/commit/00768c
[00:52:06] <begiak> apertium-html-tools: sushain97 * 62353d4: : assets/css/style.css, assets/js/translator.js and 2 other files: Add automatic URL recognition and wire up with APY https://github.com/goavki/apertium-html-tools/commit/62353d
[00:52:09] <begiak> apertium-html-tools: sushain97 * 5ef2418: : assets/js/translator.js, index.html.in: Some UI fixes and send Analytics event https://github.com/goavki/apertium-html-tools/commit/5ef241
[00:52:11] <begiak> apertium-html-tools: sushain97 * 09c1773: assets/js/translator.js: UI fix - sync textarea sizes https://github.com/goavki/apertium-html-tools/commit/09c177
[00:52:14] <begiak> apertium-html-tools: sushain97 * 355ad05: : assets/strings/eng.json, index.html.in: Legend modifications https://github.com/goavki/apertium-html-tools/commit/355ad0
[00:52:17] <begiak> apertium-html-tools: sushain97 * d2c5420: assets/js/: .eslintrc.json, persistence.js and 2 other files: Add persistence for website translation https://github.com/goavki/apertium-html-tools/commit/d2c542
[00:52:19] <begiak> apertium-html-tools: sushain97 * 56b66d3: assets/js/persistence.js: hound https://github.com/goavki/apertium-html-tools/commit/56b66d
```

Or merging a pull request:
```
[20:17:21] <begiak> apertium-html-tools: unhammer * 77dc50d: assets/js/persistence.js: The Hound prefers single quotes https://github.com/goavki/apertium-html-tools/commit/77dc50
[20:17:25] <begiak> apertium-html-tools: unhammer * b699a71: assets/js/: persistence.js, translator.js: doh, curSrc/DstLang fallback was already implemented ␍ but pick from pairs, so we don't have to trust srcLangs/dstLangs to be always aligned https://github.com/goavki/apertium-html-tools/commit/b699a7
[20:17:30] <begiak> apertium-html-tools: unhammer * a4b1593: assets/js/: persistence.js, translator.js: doh, curSrc/DstLang fallback was already implemented ␍ but pick from pairs, so we don't have to trust srcLangs/dstLangs to be always aligned https://github.com/goavki/apertium-html-tools/commit/a4b159
[20:17:32] <begiak> apertium-html-tools: unhammer * bbe17a0: assets/js/translator.js: global constant for number of translation list buttons https://github.com/goavki/apertium-html-tools/commit/bbe17a
[20:17:35] <begiak> apertium-html-tools: unhammer * a3ec64b: assets/js/: persistence.js, translator.js: Merge branch 'master' into websiteTranslationGiellatekno https://github.com/goavki/apertium-html-tools/commit/a3ec64
```

There may be more cases where this happens. Note that these are all wrapped text here, but not on IRC: this is obviously annoying behavior. The fix will most likely require begiak _collecting webhook data_ and _not posting it_ as as soon as it arrives, checking whether it is part of a larger 'series' like a PR review, sync, or merge.

@jonorthwash @sushain97 @wei2912 Anything else that is part of this behavior?

### PR DESCRIPTION
**Pull request by Androbin**
Should fix #236
See #159
@shardulc Friendly ping.

### CODE REVIEW

**Comment by shardulc** [comment]
Can you demonstrate sample behavior of your changes? (Run a local phenny, push commits to a trial repo, see IRC logs on a test channel.)

**Comment by Androbin** [comment]
> Can you demonstrate sample behavior of your changes? (Run a local phenny, push commits to a trial repo, see IRC logs on a test channel.)

I will try. But running locally gives me:

> Mailing list configuration not fully defined, aborting.

How do I need to configure `default.py` for this?

**Comment by shardulc** [comment]
It will run perfectly fine without the mailing list configuration for the scope of this issue. In the future, to test the mailing list module, take a look at what config options it accesses.

**Comment by Androbin** [comment]
screenshot

**Comment by Androbin** [comment]
During testing I found that `phenny.reply` doesn't do anything here because `origin.sender` is not well defined here. So I replaced it with a explicit call to `phenny.msg(target)`

**Comment by Androbin** [comment]
I tested as follows:
* Set up a sample repository `Androbin/phenny-example`
* Wrote a test script that pushes 5 commits at once
* Registered the webhook and forwarded the port
* Joined `#example` on `freenode.net`
* Started Phenny locally with the old code
* Ran the test script -> phenny posted 5 messages
* Restarted Phenny with the new code
* Ran the test script -> nothing happened
* Did some debugging -> `phenny.reply` uses ill-defined `origin.sender`
* Replaced `phenny.reply` with explicit `phenny.msg(target)`
* Restarted Phenny  with the new code
* Ran the test script -> first message + "you have 4 more messages"

**Comment by shardulc** [comment]
What happens when you type '.more'? I tried locally and it didn't work. Also, who has the authority to type '.more'? Can any user do it or only the admins?

**Comment by Androbin** [comment]
Okay, seems like the `target` of the webhook notifications in the channel itself.
And it is checking the `input.nick` for the user that uses `.more` and else does nothing.

**Comment by Androbin** [comment]
It occurs to me now, that the `channel` doesn't make for a good `target` [here](https://github.com/goavki/phenny/pull/276/commits/65284ca3b41d0a23eb91cc1b657e9e94eadf9574#diff-4b7c1f23394efa49c4a3faef2220fc4cR290).

**Comment by Androbin** [comment]
This seems to be a general issue with the more module.
So I thought I'd provide `admins` and `owners` with the channels `.more` that they're on.

**Comment by Androbin** [comment]
This approach seems to work just fine.
Tell me if you'd expect a different behavior.

File: modules/more.py [hunk 1]
```diff
@@ -26,27 +26,40 @@ def break_up_fn(string, max_length):
...
+    caseless_nick = input.nick.casefold()
 
-    if caseless_nick in phenny.messages.keys():
-        msg = phenny.messages[caseless_nick][0]
-        phenny.messages[caseless_nick].remove(phenny.messages[caseless_nick][0])
-        remaining = ' (' + str(len(phenny.messages[caseless_nick])) + ')' if phenny.messages[caseless_nick] else ''
-        phenny.reply(msg + remaining)
-        if not phenny.messages[caseless_nick]:
-            del phenny.messages[caseless_nick]
+    if not show_more(phenny, caseless_nick):
```
Comment (shardulc) [inline modules/more.py]: Can you move the `if not caseless_nick in ...` condition here, and not have `show_more` return anything? I think that is more readable.

Comment (Androbin) [reply modules/more.py]: Sure, gimme a moment.

**Comment by shardulc** [comment]
@Androbin Lol was just going to edit that extra colon myself. Nice job, this stuff all works as expected. Merging this and approving your task.

**Pull request merged**